src/main/java/seedu/address/model/recurfrequency/RecurFrequency.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.recurfrequency;
+
+import java.util.Arrays;
+import java.util.HashMap;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public enum RecurFrequency {
+    WEEKLY("W"), BIWEEKLY("BW"), DAILY("D"), MONTHLY("M");
+
+    public static final String DAILY_LABEL = "(Daily)";
+    public static final String WEEKLY_LABEL = "(Weekly)";
+    public static final String BIWEEKLY_LABEL = "(Biweekly)";
+    public static final String MONTHLY_LABEL = "(Monthly)";
+    public static final String UNKNOWN_LABEL = "??";
+    private static HashMap<String, RecurFrequency> frequencyMap = new HashMap<>();
+
+    static {
+        Arrays.stream(RecurFrequency.values()).forEach(option -> frequencyMap.put(option.frequency, option));
+    }
+    public static final String INVALID_RECUR_FREQUENCY_MESSAGE = String.format("Recurrence "
+            + "frequency must be one of the following: %s", frequencyMap.keySet());
+    private String frequency;
+
+    RecurFrequency(String frequency) {
+        this.frequency = frequency;
+    }
+
+    /**
+     * Gets the RecursiveFrequency corresponding to the given String frequency, if it exists. Otherwise, returns null.
+     *
+     * @param frequency The String format of the frequency
+     * @return The RecursiveFrequency corresponding to the given String frequency, if it exists.
+     * Otherwise, throws ParseException.
+     */
+    public static RecurFrequency of(String frequency) throws ParseException {
+        if (frequencyMap.containsKey(frequency)) {
+            return frequencyMap.get(frequency);
+        } else {
+            throw new ParseException(INVALID_RECUR_FREQUENCY_MESSAGE);
+        }
+    }
+
+    /**
+     * Returns the string label of the RecurFrequency.
+     *
+     * @return the string label of the RecurFrequency
+     */
+    @Override
+    public String toString() {
+        switch (this) {
+        case DAILY:
+            return DAILY_LABEL;
+        case WEEKLY:
+            return WEEKLY_LABEL;
+        case BIWEEKLY:
+            return BIWEEKLY_LABEL;
+        case MONTHLY:
+            return MONTHLY_LABEL;
+        default:
+            //Should not happen
+            return UNKNOWN_LABEL;
```

Comment:
Hmm, in that case what would you recommend for the default case?

Reply:
i was thinking of not putting a default statement here bcs it would never get evaluated anyway? or the default statement should be there no matter what?

Reply:
I believe it would be best to have a default case for all our switches. In case the programmer misses out a case, it would be made more apparent when he tries to run the code.

I initially thought of using assertFalse but it seems like its only used for tests. I tried throwing exceptions as well but toString cannot throw exceptions.

Would it be better if I renamed it to INVALID_LABEL and assigned it a value "(Invalid)"?

Reply:
oh, in that case invalid_label would be better.